addons/towing/XEH_postInit.sqf

```diff
@@ -1,14 +1,39 @@
 #include "script_component.hpp"
-["MouseButtonDown", LINKFUNC(onMouseButtonDown)] call CBA_fnc_addDisplayHandler;
+
 ["MouseButtonUp", LINKFUNC(onMouseButtonUp)] call CBA_fnc_addDisplayHandler;
 GVAR(mouseLeft) = false;
 GVAR(mouseRight) = false;
 
-GVAR(cancel) = false;
-GVAR(canAttach) = false;
+[QGVAR(ropeAttachTo), {
+    params ["_child", "_relativeAttachPos", "_rope", "_helper"];
+    TRACE_4("ropeAttachTo",_child,_relativeAttachPos,_rope,_helper);
+    _helper ropeDetach _rope;
+    [_child, _relativeAttachPos] ropeAttachTo _rope;
+    deleteVehicle _helper;
+}] call CBA_fnc_addEventHandler;
+
+[QGVAR(attachVehicles), LINKFUNC(attachVehicles)] call CBA_fnc_addEventHandler;
+[QGVAR(detachChild), LINKFUNC(detachChild)] call CBA_fnc_addEventHandler;
+
+if (!isServer) exitWith {};
 
-[QGVAR(setTowParent), {
-    params ["_parent", "_child"];
-    _child setTowParent _parent;
+[QGVAR(cleanupParent), {
+    params ["_parent"];
+    TRACE_1("cleanupParent",_parent);
+    _parent removeEventHandler ["RopeBreak", _parent getVariable QGVAR(RopeBreakEHID)];
+    _parent setVariable [QGVAR(RopeBreakEHID), -1];
+    private _parentParentHooks = _parent getVariable [QGVAR(parentHooks), []];
+    if (_parentParentHooks isEqualTo []) then {
+        TRACE_1("remove Deleted EH",_parent);
+        _parent removeEventHandler ["Deleted", _parent getVariable QGVAR(DeletedEHID)];
+        _parent setVariable [QGVAR(DeletedEHID), -1];
+    };
 }] call CBA_fnc_addEventHandler;
 
+addMissionEventHandler ["PlayerConnected", {
+    if (GVAR(allChildren) isEqualTo []) exitWith {};
+    params ["", "", "", "_jip", "_owner"];
+    if (!_jip) exitWith {};
+    TRACE_2("pushing children",_owner,GVAR(allChildren));
+    [QGVAR(setTowParentAllChildren), [GVAR(allChildren)], _owner] call CBA_fnc_ownerEvent;
+}];
```

Comment:
I mean the setup as a whole. Why go with an EH `connected` structure instead of an approach that would use `CBA_fnc_globalEventJIP`? I'm just curious.

Reply:
If I understand you correctly, for `CBA_fnc_globalEventJIP` I would need make `GVAR(allChildren)` public which is now local to server. So I push it from server only when JIP client is connected.

Reply:
The implementation I had in mind was similar to the old one, where you had a `QGVAR(setTowParent)` EH. Instead of calling it on the target only though, I would have called it as global and JIP, like so:

`[QGVAR(setTowParent), [_child, _parent], _parent] call CBA_fnc_targetEvent;`

would become

`[QGVAR(setTowParent), [_child, _parent]] call CBA_fnc_globalEventJIP;`

The JIP event would be removed upon deletion of the object in question or when the object is no longer being towed.